src/cli-constant.js

```diff
@@ -71,54 +234,71 @@ const usage = `
-                           Defaults to ./.prettierignore.
-  --stdin                  Read input from stdin.
-  --stdin-filepath         Path to the file used to read from stdin.
-  --print-width <int>      Specify the length of line that the printer will wrap on. Defaults to 80.
-  --tab-width <int>        Specify the number of spaces per indentation-level. Defaults to 2.
-  --use-tabs               Indent lines with tabs instead of spaces.
-  --no-semi                Do not print semicolons, except at the beginning of lines which may need them.
-  --single-quote           Use single quotes instead of double quotes.
-  --no-bracket-spacing     Do not print spaces between brackets.
-  --jsx-bracket-same-line  Put > on the last line instead of at a new line.
-  --trailing-comma <none|es5|all>
-                           Print trailing commas wherever possible when multi-line. Defaults to none.
-  --parser <flow|babylon|typescript|postcss|json|graphql>
-                           Specify which parse to use. Defaults to babylon.
-  --cursor-offset <int>    Print (to stderr) where a cursor at the given position would move to after formatting.
-                           This option cannot be used with --range-start and --range-end
-  --range-start <int>      Format code starting at a given character offset.
-                           The range will extend backwards to the start of the first line containing the selected statement.
-                           This option cannot be used with --cursor-offset.
-                           Defaults to 0.
-  --range-end <int>        Format code ending at a given character offset (exclusive).
-                           The range will extend forwards to the end of the selected statement.
-                           This option cannot be used with --cursor-offset.
-                           Defaults to Infinity.
-  --no-color               Do not colorize error messages.
-  --with-node-modules      Process files inside 'node_modules' directory.
-  --version or -v          Print Prettier version.
+${indent(
+  optionArray
+    .filter(option => !option.isHidden)
+    .map(createOptionUsage)
+    .join("\n"),
+  2
+)}
 
 `.slice(1); // remove leading line break
 
+function createOptionUsage(option) {
+  const threshold = 25;
+
+  let header = `--${option.name}`;
+
+  if (option.alias) {
+    header += ` or -${option.alias}`;
+  }
+
+  switch (option.type) {
+    case "boolean":
+      // do nothing
+      break;
+    case "choice":
+      header += ` <${option.choices.join("|")}>`;
+      break;
+    default:
+      header += ` <${option.type}>`;
+      break;
+  }
+
+  if (header.length >= threshold) {
+    header += "\n" + " ".repeat(threshold);
+  } else {
+    header += " ".repeat(threshold - header.length);
+  }
+
+  return (
+    header + option.description.replace(/\n/g, "\n" + " ".repeat(threshold))
+  );
+}
+
+function indent(str, spaces) {
```

Comment:
Heh, theoretically you could use prettier to format the help 😉

Reply:
`parser: "text"` 😆